emgapi/views.py

```diff
@@ -1254,143 +1300,108 @@ def list(self, request, *args, **kwargs):
     def retrieve(self, request, accession, alias,
                  *args, **kwargs):
         obj = self.get_object()
-        response = HttpResponse()
-        response['Content-Type'] = 'application/octet-stream'
-        response['Content-Disposition'] = \
-            'attachment; filename={0}'.format(alias)
         if obj.subdir is not None:
-            response['X-Accel-Redirect'] = \
-                '/results{0}/{1}/{2}'.format(
-                    obj.genome.result_directory, obj.subdir, obj.realname
-                )
+            file_path = '{0}/{1}/{2}'.format(
+                obj.genome.result_directory, obj.subdir, obj.realname
+            )
         else:
-            response['X-Accel-Redirect'] = \
-                '/results{0}/{1}'.format(
-                    obj.genome.result_directory, obj.realname
-                )
-        return response
+            file_path = '{0}/{1}'.format(
+                obj.genome.result_directory, obj.realname
+            )
+        return emg_utils.prepare_results_file_download_response(file_path, alias)
 
 
-class ReleaseViewSet(mixins.RetrieveModelMixin,
-                     emg_mixins.ListModelMixin,
-                     viewsets.GenericViewSet):
-    serializer_class = emg_serializers.ReleaseSerializer
-    queryset = emg_models.Release.objects.all()
+class GenomeSetViewSet(mixins.RetrieveModelMixin,
+                       emg_mixins.ListModelMixin,
+                       viewsets.GenericViewSet):
+    serializer_class = emg_serializers.GenomeSetSerializer
+    queryset = emg_models.GenomeSet.objects.all()
 
     filter_backends = (
         filters.OrderingFilter,
     )
 
     ordering_fields = (
-        'version',
-        'genomes_count'
+        'name',
     )
 
-    ordering = ('-version',)
+    ordering = ('-name',)
 
-    lookup_field = 'version'
-    lookup_value_regex = '[0-9.]+'
+    lookup_field = 'name'
+    lookup_value_regex = '[^/]+'
+
+    def get_queryset(self):
+        genome_set = emg_models.GenomeSet.objects.all()
+
+        if self.kwargs.get('name'):
+            genome_set = genome_set.filter(name=self.kwargs['name'])
+
+        genome_set.annotate(genome_count=Count('genome'))
```

Comment:
Would it make sense to keep a field in the catalogue for the genomes count?. I would be more performant that having to do a count each time.

Reply:
definitely.